io/src/main/scala/laika/io/runtime/RendererRuntime.scala

```diff
@@ -128,6 +128,10 @@ object RendererRuntime {
       
       val styles =  finalRoot.styles(fileSuffix) ++ getThemeStyles(themeInputs.parsedResults)
       val pathTranslator = createPathTranslator(translatorConfig, Root / "dummy", lookup).translate(_:Path)
+
+      def createDirectory (file: File): F[Unit] =
+        Sync[F].delay(file.exists || file.mkdirs()).flatMap(if (_) Sync[F].unit
```

Comment:
```suggestion
        Sync[F].blocking(file.exists || file.mkdirs()).flatMap(if (_) Sync[F].unit
```

Why not use `Files[F]` here?

Reply:
This part gets migrated in the next PR where creating and scanning directories is handled. Leaving this as it is, for a smoother merge.